infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntime.java

```diff
@@ -182,6 +193,34 @@ private void bootstrapMachine(OpenShiftMachine machine)
           .bootstrap();
   }
 
+  /**
+   * Creates OpenShift PVCs from given OpenShift environment and pre creates subpaths if needed.
+   *
+   * @param osEnv OpenShift environment
+   * @throws InfrastructureException when failed to get existing PVCs in given OpenShift namespace
+   * @throws InfrastructureException when failed to create one of given PVCs
+   * @throws InfrastructureException when any error occurs while preparing workspace's PVCs
+   */
+  @VisibleForTesting
+  void preparePVCs(OpenShiftEnvironment osEnv) throws InfrastructureException {
+    final Collection<PersistentVolumeClaim> claims = osEnv.getPersistentVolumeClaims().values();
+    final OpenShiftPersistentVolumeClaims pvcs = project.persistentVolumeClaims();
+    final Set<String> existing =
+        pvcs.get().stream().map(p -> p.getMetadata().getName()).collect(toSet());
+    final String workspaceId = getContext().getIdentity().getWorkspaceId();
+    for (PersistentVolumeClaim pvc : claims) {
+      final String[] subpaths =
```

Comment:
I'd say that it should be performed by `WorkspaceVolumesStrategy` but not by Runtime. We could rework it in the following way:
- add a method to WorkspaceVolumesStrategy interface, something like `provision`, that would just add PVC, volumes and volumeMounts;
- then Provisioner could invoke this method instead of `prepare` method;
- Runtime would invoke prepare method which would do PVC creation and pre-creation of subpaths if it is needed.

I think it'd be better because you would be able to encapsulate the logic of subpaths pre-creation into CommonPVCStrategy.

It's just proposal and I'm OK to merge current way as soon as possible.